src/box/space_def.h

```diff
@@ -42,22 +42,35 @@ extern "C" {
 
 struct space_upgrade_def;
 
+/** See space_opts::temporary. */
+enum space_opts_temporary {
+	SPACE_OPTS_NOT_TEMPORARY = 0,
+	SPACE_OPTS_TEMPORARY = 1,
+	SPACE_OPTS_META_TEMPORARY = 2,
+};
```

Comment:
Do you think `SPACE_TYPE_META` is too ambiguous? We were planning to propose support for meta-local spaces, whose metadata is persisted, but isn't replicated. Seems to me the shorter name will not be much of an improvement

Reply:
I think "meta" is just a bad name. It doesn't seem related to what these spaces can do. But I wanted to postpone public API naming until other people join the review.

If you don't mind discussing it now, then I think these spaces should be called "volatile". Because nothing is persisted. No data, no metadata. And "volatile" is an opposite of "persistent".

About "meta-local" I don't know. The problem is that there are already "local" spaces, which aren't really local. I suggest to resolve it when this patchset is finished.

However the future plans for "meta-local" definitely add points to the necessity to introduce `space_type` in C. Regardless of what will be the naming.

Reply:
IMHO introducing a new name for a type of a space like "volatile" will be confusing for the users. They already have to distinguish between "temporary", "local" & "synchronous". Adding another word to this list complicates the learning process and adds a thing people have to remember. "What is difference between 'volatile' and 'temporary' again? Which one is persisted? Let me lookup the docs" -- just another reason to waste time and mental effort. (Kinda like the difference between fiber.new & fiber.create).

The idea behind the name "meta-temporary" was to attempt to reduce this confusion by associating them with "temporary" spaces. These are after all just temporary spaces with temporary metadata and arguably the better name for them would simply be "temporary", but this renaming would be impractical.

Remembering which one of "temporary" and "meta-temporary" is more temporary is trivial in my opinion.

Reply:
All you said is entirely subjective. Personally for me "meta-temporary", "temporary" and the future "meta-local" look the same. Hence I think these names are bad. You will have to argue with the others then.

Reply:
The name is bad, the only good solution would be to break backwards compatibility and name this new type of space "temporary", while renaming the old temporary spaces to "temporary-data". Do you think this should be done in this PR? Or maybe you have other suggestions?

Reply:
done